packages/react-icons/template.js

```diff
@@ -0,0 +1,29 @@
+function template(
+	{ template },
+	opts,
+	{ imports, interfaces, componentName, props, jsx, exports }
+) {
+	const plugins = ['jsx', 'typescript']
+	const tpl = template.smart({ plugins })
+	
+	componentName.name = componentName.name.substring(3)
+	interfaceName = 'I' + componentName.name
+	combinedExport = '{ ' + interfaceName + ', ' + componentName.name + ' }'
+	exports.declaration.name = componentName.name
+
+	return tpl.ast`
+		${imports}
+
+		interface ${interfaceName} {
+			primary?: string
```

Comment:
You might want to expand this to accept more interface properties.

Reply:
Which properties could/should be needed and why? The only possible ones I can think of would be width/height but those are inherent in the selected icon.

Reply:
Some icons can be filled iirc, is this controlled by a property on the SVG, or is it it's own separate icon?

Reply:
This is different than the model we discussed earlier, and while it may not be _too_ rigid, I wonder if it's not flexible enough. The interface we've been using that works with the flexibility of css is simply to apply an optional class name. Then any arbitrary style can be applied if necessary. The first reasonable example that comes to mind is something like `margin`, though that could potentially be applied generically to the `svg` element.

If we want to go this route, I think defaulting that value to `currentcolor` would be useful because much of the time, without knowing otherwise, an icon is the same color as neighboring text.

If this is the only way to provide the primary color, or eventually secondary or fill, etc. I wonder if this will be excess work to replicate on many icons. That is the values would always need to be known and passed to each icon.

Also, I'm not sure if this work with css-in-js where the colors might be determined at runtime.

This may be a good option for some consuming patters. I suggest maintaining className support as well.

Reply:
I think clarity on the prop contract here would be helpful. Is there a method to modify the HTML attributes on the SVG itself if needed? Consider a scenario where we need to apply ARIA, add a title attribute for some reason, etc...absolutely agree on className support as well. Are these intended to work only with the Fluent UI react implementation or with others *using* React? If the latter, I think we'll want to work through the prop contract to support a broader implementation.

Reply:
They are expected to work for any React package. Assuming we define a package interface akin to:
```
export interface IFluentIconsProps {
	primary?: string;
	className?: string;
}
```
does this allow further extension to support other HTML attributes, either by product consumers or as a wider package implementation in the future?

Reply:
Further expounding on the className, is the following worth pushing as a V1 option, with the interface defined above?
```
import * as React from "react";
import { JSX } from "react-jsx";
import { IFluentIconsProps } from "./IFluentIconsProps.types";

const AccessTime24Regular = (
  iconProps: IFluentIconsProps,
  props: React.HTMLAttributes<HTMLElement>
): JSX.Element => {
  const { primaryFill = "#212121", className } = iconProps;
  return (
    <svg
      width={24}
      height={24}
     
      xmlns="http://www.w3.org/2000/svg"
      className={className}
      {...props}
    >
      <path
        d="<snip/>"
        fill={primaryFill}
      />
      <path
        d="<snip/>"
        fill={primaryFill}
      />
    </svg>
  );
};

export default AccessTime24Regular;
```
It was earlier mentioned to me that we might not want to spread all the props in this way, but I don't know another way to get the HTML attributes in.